### ISSUE TITLE
post and pre command execution not working with tmux commands

### ISSUE THREAD

**Issue by strang3r** [tmuxinator/tmuxinator#491]
I am trying to execute tmux commands via the post and pre options in my yml config file for tmuxinator.  The actions of both the pre and post arguments are being executed (verified by adding `echo test > /tmp/test.log` and checking the log file for update) but the tmux commands are ineffective. 

Here is a test config that consistently reproduces my issue:

```
# ~/.tmuxinator/cmgw.yml

name: test
root: ~/

# Optional tmux socket
# socket_name: foo

# Runs before everything. Use it to start daemons etc.
# pre: sudo /etc/rc.d/mysqld start

# Runs in each window and pane before window/pane specific commands. Useful for setting up interpreter versions.
# pre_window: rbenv shell 2.0.0-p247

# Pass command line options to tmux. Useful for specifying a different tmux.conf.
# tmux_options: -f ~/.tmux.mac.conf

# Change the command to call tmux.  This can be used by derivatives/wrappers like byobu.
# tmux_command: byobu

# Specifies (by name or index) which window will be selected on project startup. If not set, the first window is used.
# startup_window: logs

# Controls whether the tmux session should be attached to automatically. Defaults to true.
# attach: false

# Runs after everything. Use it to attach to tmux with custom options etc.
# post: tmux -CC attach -t cmgw

pre: tmux detach-client -s test
windows:
  - test01:
  - test02:
  - test03:
  - test04:
  - test05:
post: tmux move-window -s test:5 -t test:0
```
This should automatically detach any existing clients already using this session, create five windows and then move window 5 to window 0 (I have `set -g base-index 1` and `setw -g pane-base-index 1` in my .tux.conf which I prefer 99% of the time, but there are occasions I like to use window zero).  

I tried adding `tmux ls > /tmp/test.log` and it does not display any running sessions even though I do have other sessions running.  It is almost as if tmux and tmuxinator are using different sockets, but sessions started with tmuxinator are being displayed with `tmux ls` in a shell, it is only blank when executed by tmuxinator.  I also tried adding `-L default` to the tmux commands and it made no difference.

I am currently running tmuxinator 0.8.1 but have also tried 0.9 with the same result.

Here is the output of `tmuxinator debug test`:

```
#!/bin/bash

# Clear rbenv variables before starting tmux
unset RBENV_VERSION
unset RBENV_DIR

tmux start-server;

  cd /home/agall

  # Run pre command.
  tmux detach-client -s test

  # Create the session and the first window. Manually switch to root
  # directory if required to support tmux < 1.9
  TMUX= tmux new-session -d -s test -n test01


  # Create other windows.
  tmux new-window  -t test:2 -n test02
  tmux new-window  -t test:3 -n test03
  tmux new-window  -t test:4 -n test04
  tmux new-window  -t test:5 -n test05


  # Window "test01"

  # Window "test02"

  # Window "test03"

  # Window "test04"

  # Window "test05"

  tmux select-window -t 1

  if [ -z "$TMUX" ]; then
    tmux -u attach-session -t test
  else
    tmux -u switch-client -t test
  fi

tmux move-window -s test:5 -t test:0
```

Any help resolving this is appreciated!

**Comment by ethagnawl** [comment]
Hey @strang3r,

I think two different issues are at play here:

1) The `pre` command isn't run if there's an existing session with the same name as the project. See [template.erb](https://github.com/tmuxinator/tmuxinator/blob/master/lib/tmuxinator/assets/template.erb#L9-L13).
2) This may be a bug (I'll open a separate issue), but `post` doesn't seem to run on (at least on my machine) until the session is killed or detached from. When I run your sample, I see: `can't move window: index in use: 0` output upon exit or detach. (I also added some debug statements which verified that the command is actually being run on exit/detach and this message isn't an artifact.)

Fortunately, I _think_ the new project hooks which will be introduced in #486 will allow you to achieve what you're after.

**Comment by ethagnawl** [comment]
Hi, @strang3r. Are you still seeing this behavior with the most recent version of tmuxinator (0.12.0) and the new hook API?

**Comment by ethagnawl** [comment]
I'm going to close this for now. Please feel free to reopen if anyone is still seeing this issue.

**Comment by strang3r** [comment]
The on_project_start hook works perfectly for detaching existing sessions as I had originally intended, thank you.

I was not able to determine how to use the project hooks to switch windows after they have been created (as per my original attempt using post).  I was able to work around this by creating a temporary window to issue a tmux command and then immediately closing the window.  It's a but kludgy but it works:

```
on_project_start: tmux detach-client -s test
windows:
  - test01: ssh test01
  - test02: ssh test02
  - test03: ssh test03
  - test04: ssh test04
  - test-master: ssh test-master
  - post: tmux move-window -s test:5 -t test:0; exit
```

This can remain closed, I just wanted to provide an update.
